openbas-framework/src/main/java/io/openbas/asset/AssetGroupService.java

```diff
@@ -120,8 +120,11 @@ public AssetGroup computeDynamicAssets(@NotNull final AssetGroup assetGroup) {
     Specification<Endpoint> specification = computeFilterGroupJpa(assetGroup.getDynamicFilter());
     List<Asset> assets =
         this.endpointService.endpoints(specification).stream()
+            .filter(
+                endpoint ->
+                    endpoint.getAgents().getFirst().getParent() == null
+                        && endpoint.getAgents().getFirst().getInject() == null)
```

Comment:
**suggestion:** 

Can be changed to 

    Specification<Endpoint> specification = computeFilterGroupJpa(assetGroup.getDynamicFilter());
    Specification<Endpoint> specification2 = Specification.not(findEndpointsForInjection());
    List<Asset> assets =
        this.endpointService.endpoints(specification.and(specification2)).stream()
            .map(endpoint -> (Asset) endpoint)
            .distinct()
            .toList();

To improv performance